### ISSUE TITLE
Create reference to trigger in JS event objects

### ISSUE THREAD

**Issue by andy-blum** [uikit/uikit#4033]
### Summary

For WCAG compliance,  when a modal/dialog box closes, "the user's point of regard [should be] maintained by returning focus to the element that triggered display of the dialog." ([source](https://www.w3.org/TR/wai-aria-practices/examples/dialog-modal/dialog.html)) This is rather difficult to do though in cases where multiple buttons/links may open the modal, or when not using the `uk-toggle="target: selector"` patterns

What I'd _love_ to see is having the buttons that kick off the `beforeshow`, `show`, and `shown` chain of events either reference themselves in the event object, or in some other way make it clear to the `beforehide`, `hide`, and `hidden` events so that focus can be passed back to the correct opening element.

### Priority

This specific implementation is not a world-changing priority for us, but if UIKit 4 [implements a couple of key features](https://github.com/uikit/uikit/issues/3967#issuecomment-528895576) I could see it easily being a starting point for virtually every web project we do moving forward

**Comment by janschoenherr** [comment]
I think that's quite easy. Until this is done by UIkit itself you could listen for the `show` event and store the current `document.activeElement`. On `hide` you could restore the focus to it. `element.focus()`

**Comment by piotr-cz** [comment]
Its possible read target from options

```js
const toggle = UIkit.toggle($toggleButton)

// Add events to target
UIkit.util.on(toggle.$props.target, 'shown', event =>
   // Manipulate toggle buton element
)
```
